### DISCUSSION TITLE
Symbolic Regression Time Series Operator Question

### DISCUSSION THREAD

**Discussion by qizzhao** [MilesCranmer/PySR#722] [Q&A]
Lets said I have a dataset which contains N features data in T days.  My X is a matrix with dimension (T, N) , and Y is a vector with length T.   There are some basic time-series functions below:

> ma( X,  d ):       d is a random generated constant number. Calculating moving average in the past d days of each feature n in X.    X[t, n] = mean( X[t-d+1:t, n] )

> delay( X, d ):    d is a random generated constant number. Replacing X with the past d day value in each feature n.    X[t, n] = X[t-d+1, n]

> argmax(  X,  d): d is a random generated constant number. Taking the largest number in the past d day of each feature n.    X[t, n] =  max( X[t-d+1:t, n] )

I would like to put those functions above into Symbolic Regression model training and input as custom operators. 
Is that possible to achieve that? and how? Give a short example please. Thank you.

**Comment by qizzhao** [comment]
Could anyone help to answer this question above. I know how to create the binary operator and unary operator. But none of them use rolling data.

**Comment by qizzhao** [comment]
Could anyone help to answer this question above. I know how to create the binary operator and unary operator. But none of them use rolling data.

**Comment by MilesCranmer** [comment]
Sorry for the late reply. Have you checked the other discussions? There are a couple about implementing the roll operator.

**Reply by qizzhao** [reply]
Thank you  for the reply. I found that this post is similar to what I expected: https://github.com/MilesCranmer/PySR/discussions/426.  Going to study it.
But it only show the case that rolling unary operator. What if I need a rolling binary operator which input is the combination of feature value and random generated integer.
